src/topics/tools.js

```diff
@@ -27,35 +27,35 @@ module.exports = function (Topics) {
 			throw new Error('[[error:no-topic]]');
 		}
 		const canDelete = await privileges.topics.canDelete(tid, uid);
-		if (!canDelete) {
+		const topicData = await Topics.getTopicFields(tid, ['tid', 'cid', 'uid', 'deleted', 'title', 'mainPid']);
+		const data = await plugins.fireHook('filter:topic.delete', {topicData: topicData, uid:uid, isDelete:isDelete, canDelete:canDelete});
```

Comment:
The hook name should change based on `isDelete`, if isDelete is true use `filter:topic.delete` otherwise use `filter.topic.restore`.

Reply:
Also need the same logic for canRestore

Reply:
There is no canRestore privilege, and it wasn't present there before. Should I just add `canRestore:canDelete` to the hook and change the ` if` later to ` if ( (!canDelete && isDelete) || (!canRestore && !isDelete) )`?

Reply:
Yes I think that makes sense, and change the hook to 
```
const data = await plugins.fireHook(isDelete ? 'filter:topic.delete': 'filter:topic.restore', { topicData: topicData, uid: uid, isDelete: isDelete, canDelete:canDelete, canRestore: canDelete});
```
I think that will allow someone to check in a plugin and prevent a restore as well

Reply:
Done